plugins/processors/aws/ec2/ec2.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	internalaws "github.com/influxdata/telegraf/internal/config/aws"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/processors/aws/utils"
+)
+
+type (
+	EC2 struct {
+		Region    string `toml:"region"`
+		AccessKey string `toml:"access_key"`
+		SecretKey string `toml:"secret_key"`
+		RoleARN   string `toml:"role_arn"`
+		Profile   string `toml:"profile"`
+		Filename  string `toml:"shared_credential_file"`
+		Token     string `toml:"token"`
+
+		CacheTTL     internal.Duration `toml:"cache_ttl"`
+		MetricNames  []string          `toml:"metric_names"`
+		Id           string            `toml:"id"`
+		InstanceType bool              `toml:"instance_type"`
+		AmiId        bool              `toml:"ami_id"`
+		Tags         []string          `toml:"tags"`
+
+		client EC2Client
+	}
+
+	EC2Client interface {
+		DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error)
+	}
+
+	CachingEC2Client struct {
+		client  EC2Client
+		ttl     time.Duration
+		fetched time.Time
+		data    map[string]*ec2.DescribeInstancesOutput
+	}
+)
+
+func (e *CachingEC2Client) DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error) {
+	id := *input.InstanceIds[0]
+	if e.data == nil {
+		e.data = map[string]*ec2.DescribeInstancesOutput{}
+	}
+	if e.fetched.IsZero() {
+		e.fetched = time.Now()
+	}
+	if time.Since(e.fetched) >= e.ttl {
+		e.data = map[string]*ec2.DescribeInstancesOutput{}
+	}
+	if _, ok := e.data[id]; !ok {
+		response, err := e.client.DescribeInstances(input)
+		if err != nil {
+			return nil, err
+		}
+		e.data[id] = response
+	}
+	return e.data[id], nil
+}
+
+var sampleConfig = `
+  ## Amazon Region
+  region = "us-east-1"
+
+  ## Amazon Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Assumed credentials via STS if role_arn is specified
+  ## 2) explicit credentials from 'access_key' and 'secret_key'
+  ## 3) shared profile from 'profile'
+  ## 4) environment variables
+  ## 5) shared credentials file
+  ## 6) EC2 Instance Profile
+  #access_key = ""
+  #secret_key = ""
+  #token = ""
+  #role_arn = ""
+  #profile = ""
+  #shared_credential_file = ""
+
+  ## Specify the TTL for metadata lookups
+  #cache_ttl = "1h"
+
+  ## Specify the metric names to annotate with EC2 metadata
+  ## By default is configured for "cloudwatch_aws_ec2", the default output from the Cloudwatch input plugin
+  #metric_names = [ "cloudwatch_aws_ec2" ]
+
+  ## Specify the metric tag which contains the EC2 Instance ID
+  ## By default is configured for "instance_id", the default from Cloudwatch input plugin when using the InstanceId dimension
+  #id = "instance_id"
+
+  ## Enable annotating metrics with the EC2 Instance Type
+  #instance_type = true
+
+  ## Enable annotating metrics with the AMI ID
+  #ami_id = true
```

Comment:
I would probably make these a list of strings.  This would be optional though, as most of the problems with boolean options don't really apply here, namely that they often end up being interdependent and they don't scale up well.